text/0000-allow-sharing-of-LWC-across-namespaces.md

```diff
@@ -0,0 +1,230 @@
+</NamespaceBundle>
+```
+
+##### Pros
+- Leverages isExposed tag without introducing any new tag.
+- Sharing is easy as you have to edit one single file to add or remove access to the namespaces.
+
+##### Cons
+- Individual component cannot be expose to different namespaces. It is controlled at namespace level so either all or none. 
+- Learning curve as we are extending isExposed flag.
+
+
+**Alernative #2** - Granular expose control 
+Add <exposed> tag to existing LightingComponentBundle file. This tag can contain list of namespaces where this components can be accessible. 
+
+
+__Note__ : isExposed flag has higher precedence over shared flag. 
+
+|  isExposed 	|   shared   	        |   Comments  	                |
+|---	        |---	                |---	                        |
+|  True 	    |  (ignored)	        | Open to the world  	        |
+|  True 	    |  (ignored)	        | Open to the world  	        |
+|  False     	|  ! isEmpty             | Open to Limited Namespaces  	|
+|  False     	|  isEmpty	            | Open to Local Namespace  	    |
+
+```
+        <!-- LightingComponentBundle.xml -->
+<?xml version="1.0" encoding="UTF-8"?>
+<LightningComponentBundle xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>49.0</apiVersion>
+    <isExposed>false</isExposed>
+    <shared>
+        <namespaces>
+            <namespace>force</namespace>
+            <namespace>forcechatter</namespace>
+            <namespace>ui</namespace>
+        </namespaces>
+    </shared>
+</LightningComponentBundle>
+```
+
+##### Pros
+- No need to introduce new xml file at namespace level since we are using existing LightingComponentBundle file.
+- Honor isExposed flag.
+- Granular control as we can control sharing of individual LWC. 
+
+##### Cons
+- Changing access for all namespace components involves changing all LWC bundle files. Too much of work for the developer.
+- Learning curve as we are introducing new flag.
+- Complex implementation as we have to set precedence logic in the code. 
+
+## Drawbacks
+
+- implementation cost, both in term of code size and complexity
+- whether the proposed feature can be implemented in user space
+- the impact on teaching people Lightning Web Components
+- integration of this feature with other existing and planned features
+- cost of migrating existing Lightning Web Components applications (is it a breaking change?)
+
+There are tradeoffs to choosing any path. Attempt to identify them here.
+
+
+## Adoption strategy
+
+* Since we are only focusing on Internal LWC scope should be limited
+
+
+## How we teach this
+
+* Update the documentation and communicate internally about isExposed flag. 
+* Since we are only focusing on Internal LWC scope should be limited
+
+## Unresolved questions
+
+* Does compiler brings in direct and in direct dependencies ? Not sure how does sub definition plays in to this.
+* Can we do anything better than in memory map for storing the access check result ? 
+
+

```

Comment:
There's a quirky detail about access checking for on core components. If there's no `minApiVersion` specified in your bundle's meta.xml file, then your component is not exposed to external namespaces. It's only applicable to SFDC code running in core. This maybe our opportunity to improve this, by making it more explicit that you intend to be exposed to external namespaces.

Reply:
thank you good to know.